include/deal.II/base/utilities.h

```diff
@@ -1285,6 +1285,15 @@ namespace Utilities
       const std::vector<char>::const_iterator &cend,
       std::vector<T>                          &object)
     {
+      // return early if nothing to copy. this avoids
+      // dereferencing iterators if the std::vector<char> is empty,
+      // which would not be allowed.
+      if (cbegin == cend)
+        {
+          object.clear();
+          return;
+        }
```

Comment:
I would hope that this only ever happens if we had a corresponding case where we're serializing an empty vector by an empty array. But I don't think we do. Rather, I believe we serialize an empty vector by putting a zero integer into the buffer. (We could change this, but that's a separate issue.)